lib/common/client/concerns/mhv_jwt_session_client.rb

```diff
@@ -46,9 +46,20 @@ def get_session
           env = get_session_tagged
           # req_headers = env.request_headers
           res_headers = env.response_headers
+
+          # Decode the JWT token and extract the user_id from the payload
           jwt_token = res_headers['authorization'].sub('Bearer ', '')
+          decoded_token = JWT.decode jwt_token, nil, false
+          subject_id = decoded_token[0]['subjectId']
```

Comment:
Can you be sure `decoded_token` will always have an index `0` and that it has a `subjectId`, or does there need to be any error handling here?

Same question for line 58 in this file.

Reply:
@rmtolmach that's a great point. I beefed up the error handling in that function.